src/utils/dockerfileUtils.ts

```diff
@@ -0,0 +1,48 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Dockerfile, DockerfileParser, Keyword } from 'dockerfile-ast';
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import { localize } from '../localize';
+import { pathNormalize } from './pathNormalize';
+
+export interface DockerfileInfo {
+    rootFolder?: string;
+    dockerfileNameRelativeToRoot: string;
+    ports?: number[];
+}
+
+export async function parseDockerfile(rootFolderPath: string, dockerfilePath: string): Promise<DockerfileInfo> {
+    const content = await readFileContent(dockerfilePath);
+    const dockerFile = DockerfileParser.parse(content);
+    const ports: number[] = await getExposedPorts(dockerFile);
+    let dockerFilenameRelativeToRoot = path.relative(rootFolderPath, dockerfilePath);
+    dockerFilenameRelativeToRoot = pathNormalize(dockerFilenameRelativeToRoot, 'Linux')
+
+    return {
+        rootFolder: rootFolderPath,
+        dockerfileNameRelativeToRoot: dockerFilenameRelativeToRoot,
+        ports: ports
+    };
+}
+
+async function getExposedPorts(dockerFile: Dockerfile): Promise<number[]> {
+    const insts = dockerFile.getInstructions();
```

Comment:
Do we care about multi-stage Dockerfiles?  (I.e. not all stages will contribute to an end set of exposed ports.)

Reply:
I can't think of a reason why the intermediate (not part of final stage) stages would expose ports. Exposing additional ports is not blocking compose up to bring up the container. My preference is go with this until we hear from customer.